docs/understanding-the-flutterflow-ui/projects/how-to-run-test-projects.md

```diff
@@ -0,0 +1,94 @@
+---
+slug: how-to-run-test-projects
+title: How to Run and Test Projects
+tags: []
+toc_max_heading_level: 5
+---
+There are 4 ways to test your project in FlutterFlow.
+- Preview
+- Test
+- Run
+- Local Run
+
+## Preview Mode
+You can use the Preview Mode to quickly try out your app on a virtual device without waiting for it to build. This is helpful primarily for navigation and animations. You can also preview your app in the Dark/Light mode and visualize it on various mobile, tablet, and desktop devices.
+
+<div class="alert alert--info" role="alert">
+#### When to Use Preview Mode
+The main benefit is that your app will load instantly. However, because of the limitations this feature is used much less often that the other testing modes.
+</div>
+<div class="alert alert--warning" role="alert">
+  #### Preview Mode Limitations
+  - Actions may not trigger or work properly.
+  - FontAwesome icons jump around when mouse hovers over certain material widgets.
+  - Firestore data is not loaded from Firebase.
+  - Firebase auth flow can't be tested. We always allow log in.
+  - API Calls can't be run or tested here.
+  - Refresh if animation actions are not working.
+  - Refresh if Clear TextFields actions are not working
+  - RevenueCat data is not loaded.
+  - Paywall actions execute as if the entitlement is active.
+  - Hero Animation may not work on dynamically generated widgets.
+  - Dropdown disabling does not work in
+  - Preview Mode.
+  - Tooltip does not work for some screen sizes in Preview Mode.
+</div>
+
+## Test Mode
+
+Test mode uses Flutter's "Hot Reload" feature, which lets you immediately see any changes made to code in an emulator or on-device. Running your app in _Test_ Mode helps you experiment, test UIs, and fix bugs faster.
+
+To run your app in Test Mode:
+
+1. Click on the **Test icon** or press **Cmd/Ctrl + I** (a keyboard shortcut). This will run your app in a new browser window.
+2. Switch to the app builder and make some changes, such as changing colors and alignment.
+3. Now switch to the Test Mode tab and click **Instant Reload** or press **Cmd/Ctrl + J** to see the changes in under 10 seconds.
+
+<h4>Test Mode Button Colors</h4>
```

Comment:
Button Colors -> Button Indicators maybe?

Reply:
Visually the color is most prominent and would seem to be the quickest identifiable and thus searchable feature. I would vote to keep it.